va/va_test.go

```diff
@@ -734,3 +736,434 @@ func TestLogRemoteDifferentials(t *testing.T) {
 		})
 	}
 }
+
+// setup returns an in-memory VA and a mock logger. The default resolver client
+// is MockClient{}, but can be overridden.
+func setupVA(srv *httptest.Server, ua string, rvas []RemoteVA, mockDNSClient bdns.Client) (*ValidationAuthorityImpl, *blog.Mock) {
+	features.Reset()
+	fc := clock.NewFake()
+
+	mockLog := blog.NewMock()
+	if ua == "" {
+		ua = "user agent 1.0"
+	}
+
+	va, err := NewValidationAuthorityImpl(
+		&bdns.MockClient{Log: mockLog},
+		nil,
+		0,
+		ua,
+		"letsencrypt.org",
+		metrics.NoopRegisterer,
+		fc,
+		mockLog,
+		accountURIPrefixes,
+		PrimaryPerspective,
+		"ARIN",
+	)
+
+	if mockDNSClient != nil {
+		va.dnsClient = mockDNSClient
+	}
+
+	// Adjusting industry regulated ACME challenge port settings is fine during
+	// testing
+	if srv != nil {
+		port := getPort(srv)
+		va.httpPort = port
+		va.tlsPort = port
+	}
+
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create validation authority: %v", err))
+	}
+	if rvas != nil {
+		va.remoteVAs = rvas
+	}
+	return va, mockLog
+}
+
+type rvaConf struct {
+	// rir is the Regional Internet Registry for the remote VA.
+	rir string
+
+	// ua if set to pass, the remote VA will always pass validation. If set to
+	// fail, the remote VA will always fail validation with probs.Unauthorized.
+	// This is set to pass by default.
+	ua string
```

Comment:
I think this comment is not quite correct. `ua` is set to `""` by default, by the normal zero value rules.

Looking in `setupRVAs`, if it receives an empty ua it will default to `"user agent 1.0"`.

```suggestion
	// ua the user agent to be used by this remote VA. Different test cases use the
	// user agent string to accept or reject VAs selectively to create conditions for
	// testing. For instance, `TestDoDCVMPIC` accepts all requests with a user agent
	// of "pass".
	ua string
```